Circos revisions for paper

hexylena opened the pull request:
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [ ] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

Reviewers had some good suggestions that we need to fix before publication.

bgruening commented on tools/circos/circos.xml:
```diff
@@ -287,6 +314,7 @@ cp '${data.data_source}' circos/data/highlight-${hi}.txt &&
                 </conditional>
                 <param name="orientation" type="boolean" label="Orient Inwards" truevalue="in" falsevalue="out"
                     help="When yes, the plot will face inwards; i.e. lowest values will be plotted to the outside"/>
+                <param name="minsize" type="float" value="0" label="Minimum size (in chromosome units)" min="0"/>
```
maybe move this also into the macro?

bgruening commented on tools/circos/macros.xml:
```diff
@@ -164,9 +164,20 @@ $r, $g, $b, $a
+                    <option value="data">relative to data</option>
+                </param>
+                <when value="plot">
+                    <param name="y0" type="float" value="0" label="Inside Radius" help="Values are relative to the plot, 0 is the center of the plot, 1 is the outside of the plot."/>
```
add min/max?

hexylena replied:
We did this before and had to revert a bunch due to the unnecessary constraint. Negative numbers are valid, and numbers greater than 1 are valid, 1 is just the 'edge' of the plot (which may not be at the edge of the page), and you're free to plot at 1.2 if you need (and I have in occasion)

bgruening commented on tools/circos/macros.xml:
```diff
@@ -177,8 +188,20 @@ $r, $g, $b, $a
         <repeat name="backgrounds" title="Background">
-            <param name="y0" type="float" value="0" min="0" max="1" label="Inside Radius" help="Value must be between 1 (outer edge of track) and 0 (inner edge of the track)"/>
-            <param name="y1" type="float" value="1" min="0" max="1" label="Outside Radius" help="Value must be between 1 (outer edge of track) and 0 (inner edge of the track)"/>
+            <conditional name="radial_position">
```
is that the same as above?

hexylena replied:
yeah, can macro-ize it. I usually use duplication=3 as a guide to move into a macro, but no problem to do it here.

hexylena commented:
Thanks for the comments!

bgruening commented on tools/circos/macros.xml:
```diff
@@ -290,6 +381,31 @@ $r, $g, $b, $a
+    <conditional name="dynamic">
+        <param name="dynamic_select" type="select" label="Static / Dynamic Change">
+            <option value="static">Static: Change is always applied</option>
+            <option value="source">Color based on source chromosome color</option>
```
dynamic is missing here in the select

hexylena replied:
The second two are considered as 'dynamic', but, the repeated text label looked bad. I think I'll just remove `static:`

hexylena replied:
Oh, I see now.

hexylena replied:
Shouldn't that be an error in planemo? there's a conditional option, which has no select?

bgruening replied:
Yes, please fil an enhancement request :)

bgruening commented on tools/circos/stack-histogram.xml:
```diff
@@ -0,0 +1,35 @@
+	> '$output'
+	]]></command>
+	<inputs>
+		<param name="input" type="data" format="bigwig" label="BigWig files" multiple="True" help="These MUST have identical chromosomes and intervals, within their datasets. This tool is incredibly, intentionally naïve, and will crash if there are missing ranges in some bigwig files. This feels like a reasonable assumption to make (simplifies processing) at the expense of theoretical correctness, give ...
```
something is wrong here with the indentation ..

hexylena replied:
can you explain? it looks ok to me.

bgruening replied:
Its using tabs not spaces imho.

hexylena commented:
@bgruening any idea about the failures here?

hexylena commented:
ahh awesome, thanks @bgruening, never would have figured that out :)

bgruening commented:
@hexylena neat! Let me know when you need another review.

hexylena commented:
Ok, we think this is ready for review! We've made the following changes, largely based on reviewer comments from the paper submission:

- add built in karyotypes from circos which provide easiest path to get started
- comment out option to use custom cytogenic bands (basically never used, especially now with circos karyotype presets) and was confusing for users.
- Highlights are merged into the 2d data tracks (apparently there are 'highlights' and 'plots; type=highlight' that have very subtle differences.) This is the most universal option.
- Add 2d connector data type 
- Add scaling / zooming feature
- Add limits adjustments (per user issue)
- Added a tool to convert bigwigs to stacked histograms. It only supports the happy path, but this isn't a high-demand feature and it's OK enough for the limited use cases.

And some additional quality of life things:

- grep the error message out, if circos fails due to an internal circos error (xref https://github.com/galaxyproject/galaxy/pull/9690)
- fixed the built in karyotype files since upstream's are broken.
- More sane/pretty defaults.

bgruening commented on tools/circos/circos.xml:
```diff
@@ -1,14 +1,25 @@
 <tool id="circos" name="Circos" version="@WRAPPER_VERSION@">
     <description>visualizes data in a circular layout</description>
+    <xrefs>
+        <xref type="bio.tools">galactic_circos</xref>
```
great idea!

We could add, `<xref type="bioconda">circos</xref>` as well.

bgruening commented on tools/circos/circos.xml:
```diff
@@ -99,16 +107,16 @@ cp '${data.data_source}' circos/data/highlight-${hi}.txt &&
             <conditional name="ref">
                 <param name="ref_source" type="select" label="Reference Genome Source">
-                    <option value="history" selected="True">FASTA File from History</option>
+                    <option value="preset" selected="True"> Circos Presets </option>
```
any reason the have the whitespace the beginning?

bgruening commented on tools/circos/circos.xml:
```diff
@@ -125,39 +133,171 @@ cp '${data.data_source}' circos/data/highlight-${hi}.txt &&
                 -->
+                <when value="preset">
+                    <param name="preset_karyotype" type="select" label="Choose your Circos karyotype" help="In parentheses are the naming conventions for chromosomes used in each karyotype file. These are required for some plot parameters (e.g. filtering chromosomes or specifying rules). The karyotype will also be output along with your plot, for your review.">
+                        <option value="karyotype.arabidopsis.tair10.txt"> Arabidopsis tair10 (chr1,chr2,..)</option>
```
here too, I would remove the spaces at ^ and $

bgruening commented on tools/circos/circos.xml:
```diff
@@ -125,39 +133,171 @@ cp '${data.data_source}' circos/data/highlight-${hi}.txt &&
+                <param name="show_label" type="boolean" truevalue="yes" falsevalue="no" label="Show Label" checked="true"/>
+                <param type="float" value="0.075" label="Radius" name="radius_offset"/>
+                <param type="integer" value="24" label="Label Font Size" name="size"/>
+                <!--<param type="float" value="10" label="Thickness" name="thickness"/>-->
```
Shout this be still out? If so can you add comments why? So we can fix that in the next version or s?

hexylena replied:
yep! looks like bad copy paste :(

bgruening commented on tools/circos/circos.xml:
```diff
@@ -125,39 +133,171 @@ cp '${data.data_source}' circos/data/highlight-${hi}.txt &&
+
+            <section name="bands" title="Cytogenic Bands">
+                <param name="show_bands" type="boolean" truevalue="yes" falsevalue="no" label="Show Cytogenic Bands?" checked="true" help="Cytogenic bands can be defined inside the karyotype file"/>
+                <!--<param name="bands" type="data" format="tabular" optional="true" label="Cytogenetic Bands"
```
add comment?

hexylena replied:
Done!

hexylena commented:
@bgruening addressed, thanks so much for reviewing it!

shiltemann commented:
thanks for the review @bgruening!
